src/site/content/en/blog/2022-state-of-css/index.md

```diff
@@ -0,0 +1,1902 @@
+cascade or inherit. 
+
+In the following example, [BEM](http://getbem.com/naming/) naming convention
+scoping can be reversed into the actual intent. The BEM selector is attempting
+to scope the color of a `header` element to a `.card` container with naming
+conventions. This requires that the header has this classname on it, completing
+the goal. With `@scope`, no naming conventions are required in order to complete
+the same goal without marking up the header element:
+
+```css
+.card__header {
+  color: var(--text);
+}
+
+/* with @scope becomes */
+
+@scope (.card) {
+  header {
+    color: var(--text);
+  }
+}
+```
+
+Here's another example, less component-specific and more about the global scope
+nature of CSS. Dark and light themes have to coexist inside a stylesheet, where
+order matters in determining a winning style. Usually this means dark theme
+styles come after the light theme; this establishes light as the default and
+dark as the optional style. Avoid the ordering and scope battling with `@scope`:
+
+```css
+​​@scope (.light-theme) {
+  a { color: purple; }
+}
+
+@scope (.dark-theme) {
+  a { color: plum; }
+}
+```
+
+To complete the story here, `@scope` also allows the establishing of where the
+style scope ends. This can't be done with any naming convention or preprocessor;
+it's special and only something CSS built-in to the browser can do. In the
+following example, `img` and `.content` styles are exclusively applied when a
+child of a `.media-block`isand a sibling or parent of `.content`:
+
+```css
+@scope (.media-block) to (.content) {
+  img {
+    border-radius: 50%;
+  }
+
+  .content {
+    padding: 1em;
+  }
+}
+```
+
+**Resources**  
+Spec: [https://www.w3.org/TR/css-scoping-1/](https://www.w3.org/TR/css-scoping-1/)  
+Explainer: [https://css.oddbird.net/scope/explainer/](https://css.oddbird.net/scope/explainer/)  
+
+### No CSS way for a masonry layout
+
+Before CSS masonry with grid, JavaScript was the best way to achieve a masonry
+layout, as any of the CSS methods with columns or flexbox would inaccurately
+represent the content order.
+
+After CSS masonry with grid, no JavaScript libraries will be required and the
+content order will be correct.
+
+<figure data-size="full">
+  {% Img 
+    src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/xfOiyctwsXa8wnfkuoS2.png", 
+    alt="Screenshot of the masonry layout which shows numbers traveling along the top, then going down.", 
+    width="800", 
+    height="547" 
+  %}
+  <figcaption>
+    Image and demo from Smashing Magazine<br>
+    <a href="https://www.smashingmagazine.com/native-css-masonry-layout-css-grid/">https://www.smashingmagazine.com/native-css-masonry-layout-css-grid/</a>
+  </figcaption>
+</figure>
+
+The preceding demo is achieved with the following CSS:
+
+```css/3
+.container {
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
+  grid-template-rows: masonry;
+}
+```
+
+It's still under debate whether or not CSS grid is the best place for this
```

Comment:
Not sure this is under debate at this point, I think that ship may have sailed.

Reply:
cool, will update